react/src/lib/components/SubsurfaceViewer/layers/labeled_points/labeledPointsLayer.ts

```diff
@@ -0,0 +1,146 @@
+import { CompositeLayer, UpdateParameters } from "@deck.gl/core/typed";
+import { ScatterplotLayer } from "@deck.gl/layers/typed";
+import { isEqual } from "lodash";
+
+import {
+    ExtendedLayerProps,
+    invertZCoordinate,
+    defineBoundingBox,
+} from "../utils/layerTools";
+
+export interface LabeledPointsLayerProps<D> extends ExtendedLayerProps<D> {
+    /**
+     * Point positions as [x, y, z, x, y, z....].
+     */
+    pointsData: number[];
+
+    /**
+     * Point color defined as RGB or RGBA array. Each component is in 0-255 range.
+     */
+    color: [number, number, number] | [number, number, number, number];
+
+    /**
+     * The units of the point radius, one of `'meters'`, `'common'`, and `'pixels'`.
+     */
+    radiusUnits: "meters" | "common" | "pixels";
+
+    /**
+     * Point radius defined in radius units.
+     */
+    pointRadius: number;
+
+    /**  If true means that input z values are interpreted as depths.
+     * For example depth of z = 1000 corresponds to -1000 on the z axis. Default true.
+     */
+    ZIncreasingDownwards: boolean;
+
+    // Non public properties:
+    setReportedBoundingBox?: React.Dispatch<
+        React.SetStateAction<[number, number, number, number, number, number]>
+    >;
+}
+
+const defaultProps = {
+    "@@type": "LabeledPointsLayer",
```

Comment:
Is this needed?

Reply:
Any other layers have @@type in their default props. If this isn't needed let's remove @@type field in all the default props.

Reply:
Ok, then I might be wrong :)